force-app/main/default/classes/GiftBatch.cls

```diff
@@ -211,4 +260,22 @@ public inherited sharing class GiftBatch implements IGiftBatch {
         }
         set;
     }
+
+    private Boolean hasFinishedProcessing() {
+        return this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.ABORTED.name()
+            || this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.FAILED.name()
+            || this.asyncApexJob.Status == UTIL_BatchJobService.ApexJobStatus.COMPLETED.name();
+    }
+
+    private Boolean hasBeenProcessed() {
+        return this.batch?.Latest_Apex_Job_Id__c != null;
+    }
+
+    private AsyncApexJobId lastProcessedJobId() {
+        return new AsyncApexJobId(this.batch.Latest_Apex_Job_Id__c);
```

Comment:
What happens if this field contains something that is not an Apex job ID?

Reply:
Since we're using the AsyncApexJobid type, an exception will be thrown which should halt the operation.